File: JAVA/Pertemuan 01/src/id/ac/uph/pbo/pertemuan01/Tas.java
package id.ac.uph.pbo.pertemuan01;

public class Tas {
    private String merek;
    private String warna;
    private double kapasitasLiter;
    private int    jumlahKantong;
    private boolean terkunci;

    public Tas(String merek, String warna, double kapasitasLiter, int jumlahKantong) {
        this.merek          = merek;
        this.warna          = warna;
        this.kapasitasLiter = kapasitasLiter;
        this.jumlahKantong  = jumlahKantong;
        this.terkunci       = false;
    }

    public void masukkanBarang(String barang) {
        System.out.println("[Tas " + merek + "] Memasukkan: " + barang);
    }

    public void keluarkanBarang(String barang) {
        System.out.println("[Tas " + merek + "] Mengeluarkan: " + barang);
    }

    public void kunci() {
        terkunci = true;
        System.out.println("[Tas " + merek + "] Terkunci 🔒");
    }

    public void tampilkanInfo() {
        System.out.println("  Tas       : " + merek + " (" + warna + ")");
        System.out.println("  Kapasitas : " + kapasitasLiter + " Liter");
        System.out.println("  Kantong   : " + jumlahKantong);
        System.out.println("  Terkunci  : " + (terkunci ? "Ya" : "Tidak"));
    }
}

File: PYTHON/Pertemuan 01/Tas.py
import sys
sys.stdout.reconfigure(encoding='utf-8')
class Tas:
    def __init__(self, merek: str, warna: str, kapasitas_liter: float, jumlah_kantong: int):
        self.merek = merek
        self.warna = warna
        self.kapasitas_liter = kapasitas_liter
        self.jumlah_kantong = jumlah_kantong
        self.terkunci = False

    def masukkan_barang(self, barang: str):
        print(f"[Tas {self.merek}] Memasukkan: {barang}")

    def keluarkan_barang(self, barang: str):
        print(f"[Tas {self.merek}] Mengeluarkan: {barang}")

    def kunci(self):
        self.terkunci = True
        print(f"[Tas {self.merek}] Terkunci 🔒")

    def tampilkan_info(self):
        print(f"  Tas       : {self.merek} ({self.warna})")
        print(f"  Kapasitas : {self.kapasitas_liter} Liter")
        print(f"  Kantong   : {self.jumlah_kantong}")
        print(f"  Terkunci  : {'Ya' if self.terkunci else 'Tidak'}")

